sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -261,6 +261,14 @@ partitionVal
     : identifier (EQ constant)?
     ;
 
+dropPartitionSpec
+    : PARTITION '(' dropPartitionVal (',' dropPartitionVal)* ')'
+    ;
+
+dropPartitionVal
+    : identifier (comparisonOperator constant)?
```

Comment:
It has to be in this format? `partCol1 > 2` How about `2 > partCol1`?

Reply:
yes, in Hive it has to be like this. `2 > partCol1` is not supported by Hive.

Reply:
Hive does throw an error in that case, you mean asking that error is a parsing or another kind of exception?

Reply:
yea, yes. I like user-understandable error messages.

Reply:
Hive throws this parser exception:
```
hive> alter table test1 drop partition(1 > c);
NoViableAltException(368@[])
	at org.apache.hadoop.hive.ql.parse.HiveParser_IdentifiersParser.identifier(HiveParser_IdentifiersParser.java:12014)
	at org.apache.hadoop.hive.ql.parse.HiveParser_IdentifiersParser.dropPartitionVal(HiveParser_IdentifiersParser.java:11684)
	at org.apache.hadoop.hive.ql.parse.HiveParser_IdentifiersParser.dropPartitionSpec(HiveParser_IdentifiersParser.java:11563)
	at org.apache.hadoop.hive.ql.parse.HiveParser.dropPartitionSpec(HiveParser.java:44851)
	at org.apache.hadoop.hive.ql.parse.HiveParser.alterStatementSuffixDropPartitions(HiveParser.java:11564)
	at org.apache.hadoop.hive.ql.parse.HiveParser.alterTableStatementSuffix(HiveParser.java:8000)
	at org.apache.hadoop.hive.ql.parse.HiveParser.alterStatement(HiveParser.java:7450)
	at org.apache.hadoop.hive.ql.parse.HiveParser.ddlStatement(HiveParser.java:4340)
	at org.apache.hadoop.hive.ql.parse.HiveParser.execStatement(HiveParser.java:2497)
	at org.apache.hadoop.hive.ql.parse.HiveParser.statement(HiveParser.java:1423)
	at org.apache.hadoop.hive.ql.parse.ParseDriver.parse(ParseDriver.java:209)
	at org.apache.hadoop.hive.ql.parse.ParseUtils.parse(ParseUtils.java:74)
	at org.apache.hadoop.hive.ql.parse.ParseUtils.parse(ParseUtils.java:67)
	at org.apache.hadoop.hive.ql.Driver.compile(Driver.java:615)
	at org.apache.hadoop.hive.ql.Driver.compileInternal(Driver.java:1829)
	at org.apache.hadoop.hive.ql.Driver.compileAndRespond(Driver.java:1776)
	at org.apache.hadoop.hive.ql.Driver.compileAndRespond(Driver.java:1771)
	at org.apache.hadoop.hive.ql.reexec.ReExecDriver.compileAndRespond(ReExecDriver.java:126)
	at org.apache.hadoop.hive.ql.reexec.ReExecDriver.run(ReExecDriver.java:214)
	at org.apache.hadoop.hive.cli.CliDriver.processLocalCmd(CliDriver.java:239)
	at org.apache.hadoop.hive.cli.CliDriver.processCmd(CliDriver.java:188)
	at org.apache.hadoop.hive.cli.CliDriver.processLine(CliDriver.java:402)
	at org.apache.hadoop.hive.cli.CliDriver.executeDriver(CliDriver.java:832)
	at org.apache.hadoop.hive.cli.CliDriver.run(CliDriver.java:770)
	at org.apache.hadoop.hive.cli.CliDriver.main(CliDriver.java:694)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.hadoop.util.RunJar.run(RunJar.java:221)
	at org.apache.hadoop.util.RunJar.main(RunJar.java:136)
FAILED: ParseException line 1:33 cannot recognize input near '1' '>' 'c' in drop partition statement
```
so yes, it is analogous to this.

Reply:
thanks for the check. I still like meaningful messages though, we shold wait for other reviewer's comments.